### PR TITLE
fix(scripts): stop the link audit calling hyphen variants a wrong artist

### PR DESCRIPTION
**Pull request by BreakableHoodie**
## Summary

A production run of the identity audit flagged both of #171's links as MISMATCH:

```
MISMATCH #171 spotify      db="Kman & the 45s" platform="K-Man & The 45s"
MISMATCH #171 apple_music  db="Kman & the 45s" platform="K-Man & The 45s"
```

That is the **same band** — the Apple Music slug is literally `k-man-the-45s`.

`normalise()` replaces every non-alphanumeric run with a space, so the hyphen inside `K-Man` splits the token (`"k man"`) and no longer equals its unpunctuated twin (`"kman"`). `containsTokens` can't bridge it either, so it fell through to MISMATCH.

**This is not cosmetic.** MISMATCH is the actionable bucket and exits 1. The script's own header says billing variants *"belong in REVIEW, never in MISMATCH, or the signal drowns"* — it was drowning its own signal, in the one category that means fans are being sent to a stranger's music.

Refs #788

## What changed

| File | Change |
|------|--------|
| `scripts/audit-streaming-link-identity.mjs` | Equality-after-despacing → `OK`; export the pure helpers; guard `main()`; surface `err.stderr`/`err.code` |
| `scripts/__tests__/auditStreamingLinkIdentity.test.js` | New — 13 cases against `classify()` |
| `vitest.config.js` | Extend test `include` to `scripts/**/__tests__/**` |

The despace check is **equality, never containment** — containment would collapse genuinely different names, which is what `containsTokens`'s own doc comment warns about.

## Three defects found while verifying, beyond the reported one

1. **The test would never have run.** Root vitest `include` was `functions/**/__tests__/**` only, so a test under `scripts/` was collected by nothing and `npm test` passed by finding zero files. Extended the pattern. Coverage stays scoped to `functions/**/*.js` via `coverage.include`, so a script joining the test set does not move the ratchet denominator. Backend files **115 → 116** and tests **1149 → 1162** confirm it is now genuinely collected.

2. **The script was untestable by construction.** It exported nothing and called `main()` at module scope, so importing it from a test would have executed the whole audit against production D1 and the network. Now exports the pure helpers behind an entry-point guard.

3. **It swallowed the real error.** The handler printed only `err.message`, so a transient wrangler failure rendered as a bare `Command failed: <command>` with no reason — hit during this work, and it took reproducing the `execFile` call by hand to see the cause. A swallowed error in a script whose entire job is surfacing problems. Now prints `err.stderr` and `err.code`.

## Verification

- [x] **Mutation-proven, not assumed**: removing the despace line fails the K-Man case (1 failed / 12 passed); restoring it passes 13/13
- [x] End-to-end against production: **MISMATCH 2 → 0**, exit **1 → 0**, `#171` apple_music moved MISMATCH → OK
- [x] CLI behaviour unchanged: no-args → exit 2, invalid db name → exit 2, `node --eval` import no longer throws
- [x] Tests: backend 1162 passed | 6 todo (116 files); frontend 1052 passed (94 files) (`make gate`, exit 0)
- [x] ESLint 0 errors · Format check clean · Build green
- [x] CodeRabbit (`make review`): 1 finding (`process.argv[1]` undefined under `--eval`), fixed and both paths re-verified

**Honest caveat on the headline number:** `#171`'s spotify link came back **UNRESOLVED** on the final run — Spotify's oEmbed returned no name, almost certainly throttling after three audits in quick succession. So "MISMATCH 0" is partly luck on that one link; the apple_music case moving MISMATCH → OK is the load-bearing evidence.

## Not in scope

Four profiles (GZ 68, Dead Karma 108, Azathoth Entombed 132, A Dallas Welcome 234) now have **no** Spotify link — the wrong ones were nulled, which is #788's step 1 and already done. Finding the correct ones is step 2: research plus a production data write about real musicians, requiring identity proven by discography rather than name match. Deliberately left for a separate, human-reviewed pass.

## Security / correctness notes

None. Read-only tooling and a test-config include pattern. No schema, auth, API, or production data touched — the audit runs a single `SELECT`.

---
Built by Otto · Reviewed by Theo · 🤖 [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved streaming-link identity checks to recognize names that differ only by hyphenation, spacing, or article formatting.
  - Preserved accurate handling of partial matches for review, mismatches, and missing platform names classified as unresolved.
  - Enhanced command-line error reporting with clearer details and exit statuses.

- **Tests**
  - Added coverage for exact matches, normalized names, hyphenation variants, review cases, mismatches, and unresolved results.

### CODE REVIEW

File: scripts/audit-streaming-link-identity.mjs [hunk 1]
```diff
@@ -127,16 +128,25 @@ function withoutArticle(s) {
...
   if (a === b) return "OK";
+  // Spacing-only difference: normalise() turns every non-alphanumeric run into
+  // a space, so intra-word punctuation splits a token ("K-Man" -> "k man") and
+  // no longer equals its unpunctuated twin ("Kman"). Real case, #171: our
+  // "Kman & the 45s" vs the platform's "K-Man & The 45s", whose Apple slug is
+  // literally k-man-the-45s. Equality-after-despacing (never containment,
+  // which would collapse genuinely different names) folds that back to OK, so
+  // it does not pollute MISMATCH — the one bucket that means fans are being
+  // sent to a stranger's music.
+  if (a.replace(/ /g, "") === b.replace(/ /g, "")) return "OK";
```
[the remark this answers is not in the document]
Comment (BreakableHoodie) [reply scripts/audit-streaming-link-identity.mjs]: Good catch — this was a defect in my own change, and a bad one: a fix that reintroduces the bug class one layer down.

Applied. `normaliseIgnoringHyphens()` now strips only hyphens (ASCII plus U+2010–U+2015, since platform metadata does use them) from the raw name before normalising, so `K-Man` still meets `Kman` while `Sea Lion` and `Seal Ion` stay apart.

Added your exact example as a MISMATCH case, and mutation-proved it in both directions: reverting to despace-all fails `Sea Lion`, removing the hyphen check fails `K-Man`, correct passes 14/14. The rule can no longer be widened back without a red test.
